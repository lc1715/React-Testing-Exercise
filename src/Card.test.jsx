import { render } from "@testing-library/react";
import { it, expect } from "vitest";

import Card from './Card'


it('should render without crashing', () => {
    render(<Card caption='caption-test' src='http://image-test' currNum={1} totalNum={3} />)
})

it('should match snapshot', () => {
    const { asFragment } = render(
        <Card
            caption='caption-test'
            src='http://image-test'
            currNum={1}
            totalNum={3}
        />)

    expect(asFragment()).toMatchSnapshot()
})

it('should match snapshot with different props', () => {
    const { asFragment } = render(
        <Card
            caption='caption-test-2'
            src='http://image-test-2'
            currNum={1}
            totalNum={5}
        />)
    expect(asFragment()).toMatchSnapshot()
})
