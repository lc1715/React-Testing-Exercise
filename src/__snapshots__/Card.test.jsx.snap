// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      caption-test
    </h4>
    <img
      alt="caption-test"
      class="Card-image"
      src="http://image-test"
    />
    <small
      class="Card-small"
    >
      Image 1 of 3.
    </small>
  </div>
</DocumentFragment>
`;

exports[`should match snapshot with different props 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      caption-test-2
    </h4>
    <img
      alt="caption-test-2"
      class="Card-image"
      src="http://image-test-2"
    />
    <small
      class="Card-small"
    >
      Image 1 of 5.
    </small>
  </div>
</DocumentFragment>
`;
